qa/suites/orch/cephadm/upgrade/3-upgrade/staggered.yaml

```diff
@@ -0,0 +1,78 @@
+tasks:
```

Comment:
As I understood from reading the docs we are adding different types of staggered upgrade:

- by daemon_types
- by services
- by hosts
- by limit

Instead of having all of them in the same (big) test case I think it would be better to have different (smaller) test cases for each one of them. That would make it easier to debug issues (in case of future failures) and would make it easier for anybody trying to read/understand the test case.

Reply:
the reason I wanted one big test here is because I didn't want to have a bunch of different tests. Teuthology tests require a decent more actual lab resources than unit tests so it's preferable to have one test for the new feature rather than  adding 4 different tests imo.

Reply:
I see, I'm afraid it would be hard to debug failures on this test-case being so long. Anyway in the worst case we can modify it during the debugging to keep only the failing part :+1: